botocore/httpsession.py

```diff
@@ -223,9 +229,7 @@ def __init__(self, proxies=None, proxies_settings=None):
     def proxy_url_for(self, url):
         """Retrieves the corresponding proxy url for a given url."""
         parsed_url = urlparse(url)
-        proxy = self._proxies.get(parsed_url.scheme)
-        if proxy:
-            proxy = self._fix_proxy_url(proxy)
+        proxy = self.proxy_url_for_scheme(parsed_url.scheme)
```

Comment:
As there is no further checks/actions on proxy, it can be returned straight away.
`return self.proxy_url_for_scheme(parsed_url.scheme)`